src/shim.c

```diff
@@ -954,6 +966,51 @@ connect_to_proxy(struct cc_shim *shim)
 	return false;
 }
 
+inline void sleep_ms(int ms)
+{
+	struct timespec ts = { 0, ms * 1000000L };
+	nanosleep(&ts, NULL);
+}
+
+/*
+ * Try to re-establish tcp/unix connection with proxy in shim->timeout seconds.
+ *
+ * \param shim \ref cc_shim
+ *
+ * \return true on success, false on failure
+ */
+bool reconnect_to_proxy(struct cc_shim *shim)
+{
+	shim_warning("Reconnecting to cc-proxy (timeout %d s)\n",
+			shim->timeout);
+	int time = 0;
+
+try_again:
+	sleep_ms(RECONNECT_POLL_MS);
+	time += RECONNECT_POLL_MS;
+	if (time >= shim->timeout * 1000) {
```

Comment:
Interesting. For me it's completely opposite :) Nevertheless, I'm not sure it's a good idea to mix double/float literals with integer variables.